metadb/api/dbimport.py

```diff
@@ -134,6 +137,92 @@ def registerReference(self, guid, reference_set_id, name, length):
 
         return reference
 
+    def registerFieldSet(self, guid, reader, description=None):
+        fieldSet = self.session.query(FieldSet).filter(FieldSet.guid == guid).first()
+        fieldMap = {}
+
+        if fieldSet is None:
+            try:
+                fieldSet = FieldSet(guid=guid, description=description)
+                self.session.add(fieldSet)
+                self.session.commit()
+
+            except exc.DataError as e:
+                self.session.rollback()
+                raise ValueError("{0} : {1} ".format(str(e), guid))
+
```

Comment:
Can we add an if statement if( reader) surrounding all the reader statements? ie. allow calling registerFieldSet with reader=None.